simdjson/__init__.pyi

```diff
@@ -0,0 +1,108 @@
+import json
+from typing import (
+    AbstractSet,
+    Any,
+    Dict,
+    Final,
+    Iterator,
+    List,
+    Literal,
+    MutableMapping,
+    NewType,
+    Optional,
+    Tuple,
+    Union,
+    ValuesView,
+)
+
+
+Primitives = Union[int, float, str, boolean]
+K = str
+V = Union['Object', 'Array', Primitives]
+
+
+class Object(Mapping[K, V]):
+  
+    def __getitem__(self, key: K) -> V:
+        ...
+
+    def __iter__(self) -> Iterator[V]:
+        ...
+
+    def __len__(self) -> int:
+        ...
+
+    def as_dict(self) -> Dict[K, V]:
+        ...
+
+    def at_pointer(self, key: str) -> V:
+        ...
+
+    def get(self, key: K, default: Optional[V] = None) -> V:
+        ...
+
+    def keys(self) -> AbstractSet[K]:
+        ...
+
+    def values(self) -> ValuesView[V]:
+        ...
+
+    def items(self) -> AbstractSet[Tuple[K, V]]:
+        ...
+
+    @property
+    def mini(self) -> 'Object':
+        ...
+
+
+Buffer = NewType('Buffer', bytes)
+
+
+class Array(Sequence[V]):
+    def count(self, v: V) -> int:
+        ...
+
+    def as_list(self) -> List[V]:
+        ...
+
+    def as_buffer(self, *, of_type: Literal['d', 'i', 'u']) -> Buffer:
+        ...
+
+    def at_pointer(self, key: str) -> V:
+        ...
+
+    def index(
+        self,
+        x: V,
+        start: Optional[int] = None,
+        end: Optional[int] = None,
+    ) -> int:
+        ...
+
+    @property
+    def mini(self) -> 'Array':
+        ...
+
+    @property
+    def slots(self) -> int:
+        ...
+
+
+class Parser:
+    def __init__(self, max_capacity: int = ...) -> None:
+        ...
+
+    def load(self, path: str, recursive: bool = False) -> Union[Array, Object]:
```

Comment:
This (and parse) should probably be `Union[V, dict, list]` since `parse('5')` is valid JSON, and normal objects will be returned instead of proxies if `recursive=True`.

Reply:
right, missed that. I think it was even in docs.
we can actually do `@overload`, so will something like so be more accurate:

```
def load(self, path: str, recursive: bool = False) -> Union[Array, Object, V]:
@overload
def load(self, path: str, recursive: bool = True) -> V:
```

Reply:
I believe it should be correct right now.